src/FreeformIdeas.js

```diff
@@ -94,11 +177,50 @@ const FreeFormIdeas = () => {
         <Idea
           key={key}
           onEdit={editIdea}
-          onSelect={setSelected}
-          selected={idea.id === selectedIdea.id}
+          onSelect={setSelectedId}
+          selected={idea.id === selectedIdea?.id}
+          linkStart={onLinkStart}
+          linkDestination={onLinkEnd}
```

Comment:
Let's keep the convention of naming props that are callback functions `onEvent`. We should also refer to the functions as `handleEvent` unless there's a better name. In this case I'd like `onLinkStart={handleLinkStart}` and `onLinkEnd={handleLinkEnd}`.